mars/scheduler/__main__.py

```diff
@@ -34,15 +34,25 @@ def config_args(self, parser):
         parser.add_argument('--nproc', help='number of processes')
 
     def create_pool(self, *args, **kwargs):
-        self._service = SchedulerService(
-            advertise_addr=self.args.advertise,
-        )
+        self._service = SchedulerService()
         self.n_process = int(self.args.nproc or resource.cpu_count())
         kwargs['distributor'] = MarsDistributor(self.n_process, 's:h1:')
         return super(SchedulerApplication, self).create_pool(*args, **kwargs)
 
+    def create_scheduler_discoverer(self):
+        advertise_endpoint = self.args.advertise or self.endpoint
+        if ':' not in advertise_endpoint:
+            advertise_endpoint += ':' + self.endpoint.rsplit(':', 1)[-1]
```

Comment:
Is it confirmed that ':' is in `self.endpoint`?

Reply:
``create_scheduler_discoverer`` is called after the actor pool is created, and ``self.endpoint`` is generated by code then, not user input.